src/MySqlConnector/MySqlClient/Types/TypeMapper.cs

```diff
@@ -0,0 +1,141 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+using MySql.Data.Serialization;
+
+namespace MySql.Data.MySqlClient.Types
+{
+	internal class TypeMapper
+	{
+		internal static TypeMapper Mapper = new TypeMapper();
+
+		private TypeMapper()
```

Comment:
Here are the beginnings of a Type Mapper.  I think that we can evolve this namespace (in further commits) to take care of most CLR Type / Column Type / DbType conversions.